packages/11ty/_plugins/figures/figure.js

```diff
@@ -0,0 +1,43 @@
+const { getPrintImage, isCanvas, isImageService } = require('./helpers')
+const tileImage = require('./iiif/tile-image')
+const transform = require('./transform')
+const Manifest = require('./iiif/manifest')
+const ManifestWriter = require('./iiif/manifest/writer')
+
+module.exports = class Figure {
+  constructor(eleventyConfig, data) {
+    this.iiifConfig = eleventyConfig.globalData.iiifConfig
+    this.data = {
+      ...data,
+      isCanvas: isCanvas(data),
+      isImageService: isImageService(data),
+      printImage: getPrintImage(eleventyConfig, data)
```

Comment:
Using the class pattern each of these might be to level properties rather than nested as properties of `this.data`.